lib/access_check_filter.js

```diff
@@ -64,6 +67,12 @@ module.exports = (hyper, req, next, options, specInfo) => {
                 }
 
                 return contentPromise.then(content => redirect(content, location, options));
+            } else if (content.status === 302) {
+                // Parsoid.js emits a redirect when the new content is parsed because
+                // there could be a race condition with updating a restriction table.
+                // If the client doesn't want a redirect - remove it.
```

Comment:
How can we know for sure that the client doesn't want a redirect here simply based on the status code (which implies otherwise) ?

Reply:
This is inside an else-if clause, if client wanted a redirect or we wanted to provide it - we wouldn've going into the if clause and never even evaluated this condition comparing the status to 302